ts/Extensions/MouseWheelZoom/MouseWheelZoom.ts

```diff
@@ -66,198 +70,257 @@ const optionsToObject = (
  * @private
- */
-const zoomBy = function (
+*/
+const waitForAutomaticExtremes = function (
+    axis: Axis
+) : void {
+    const axisOptions = axis.options;
+
+    // Options interfering with yAxis zoom by setExtremes() returning
+    // integers by default.
+    if (defined(wheelTimer)) {
+        clearTimeout(wheelTimer);
+    }
+
+    if (!defined(startOnTick)) {
+        startOnTick = axisOptions.startOnTick;
+        endOnTick = axisOptions.endOnTick;
+    }
+
+    // Temporarily disable start and end on tick, because they would
+    // prevent small increments of zooming.
+    if (startOnTick || endOnTick) {
+        axisOptions.startOnTick = false;
+        axisOptions.endOnTick = false;
+    }
+    wheelTimer = setTimeout((): void => {
+        if (defined(startOnTick) && defined(endOnTick)) {
+            // Repeat merge after the wheel zoom is finished, #19178
+            axisOptions.startOnTick = startOnTick;
+            axisOptions.endOnTick = endOnTick;
+
+            // Set the extremes to the same as they already are, but now
+            // with the original startOnTick and endOnTick. We need
+            // `forceRedraw` otherwise it will detect that the values
+            // haven't changed. We do not use a simple yAxis.update()
+            // because it will destroy the ticks and prevent animation.
+            const { min, max } = axis.getExtremes();
+            axis.forceRedraw = true;
+            axis.setExtremes(min, max);
+            startOnTick = endOnTick = void 0;
+        }
+    }, 400);
+};
+
+/**
+* Calulate the ratio of mouse position on the axis to it's length. If mousePos
+* doesn't exist, returns 0.5;
+* @private
+*/
+const calucateFixToDirection = function (
     chart: Chart,
-    howMuch: number,
-    centerXArg: number,
-    centerYArg: number,
-    mouseX: number,
-    mouseY: number,
-    options: MouseWheelZoomOptions
-): boolean {
-    const xAxis = chart.xAxis[0],
-        yAxis = chart.yAxis[0],
-        yOptions = yAxis.options,
-        type = pick(options.type, chart.zooming.type, ''),
-        zoomX = /x/.test(type),
-        zoomY = /y/.test(type);
+    axis: Axis,
+    mousePos: number
+) : number {
+    const fixToDir = mousePos ? ((mousePos - axis.pos) / axis.len) : 0.5,
+        isXAxis = axis.isXAxis;
+
+    if (isXAxis && (!axis.reversed !== !chart.inverted) ||
+            !isXAxis && axis.reversed) {
+        // We are taking into account that xAxis automatically gets
+        // reversed when chart.inverted
+        return 1 - fixToDir;
+    }
 
-    let hasZoomed = false;
+    return fixToDir;
+};
 
-    if (defined(xAxis.max) && defined(xAxis.min) &&
-        defined(yAxis.max) && defined(yAxis.min) &&
-        defined(xAxis.dataMax) && defined(xAxis.dataMin) &&
-        defined(yAxis.dataMax) && defined(yAxis.dataMin)) {
+/**
+* Check if zoom is out of the current axis, if not, change the extremes to
+* previously calulated values based on how much zoom should be applied.
```

Comment:
The method description should focus more on WHAT the method is doing not on HOW it is doing it. So just saying something like `Perform the zooming on the passed axis` should be enough ;)